src/control/lib/spdk/nvme.go

```diff
@@ -120,19 +121,35 @@ func (n *Nvme) Discover() ([]Controller, error) {
 }
 
 // Format device at given pci address, destructive operation!
-func (n *Nvme) Format(ctrlrPciAddr string) error {
+func (n *Nvme) Format(log logging.Logger, ctrlrPciAddr string) error {
 	csPci := C.CString(ctrlrPciAddr)
 	defer C.free(unsafe.Pointer(csPci))
 
-	failLocation := "NVMe Format(): C.nvme_format"
+	failLocation := "NVMe Format(): C.nvme_"
+
+	// attempt wipe of namespace #1 LBA-0
+	log.Debugf("attempting quick format on %s\n", ctrlrPciAddr)
+	retPtr := C.nvme_wipe_first_ns(csPci)
+	if retPtr == nil {
+		return errors.Errorf("%swipe_first_ns() unexpectedly returned NULL",
+			failLocation)
+	}
+	if retPtr.rc == 0 {
+		return nil // quick format succeeded
+	}
+
+	log.Errorf("%swipe_first_ns() failed, rc: %d, %s",
+		failLocation, retPtr.rc, C.GoString(&retPtr.err[0]))
 
-	retPtr := C.nvme_format(csPci)
+	// fall back to full controller format if quick format failed
+	log.Infof("falling back to full format on %s\n", ctrlrPciAddr)
+	retPtr = C.nvme_format(csPci)
```

Comment:
I'm of two minds about this approach. On the one hand, I like the "just get it done" interpretation of the admin's intent. They want the device to be formatted; they don't care that it takes a low-level format to make it happen. On the other hand, knowing as we do now that low-level formats come with some cost to the device's lifespan, I'm a little concerned that we may inadvertently cause premature wear due to use cases or scenarios that we haven't considered.

Perhaps as a compromise between making this overly complex or blindly trusting that it's the right thing to do, we can work with the test team to add a check to the harness which fails a test run if the above message is detected in the log output. In theory, it should never show up except possibly when a brand-new drive is added to the test cluster. This way we can have some confidence that we have correct assumptions about when the fallback logic is triggered.

Reply:
nvme_format should be okay now, since a previous PR removed the "user data erase" option from the format request. User data erase was the part causing the trouble in NAND drives.

Reply:
That seems reasonable to me, however the fact that you qualified your position make me think that there may still be some value in testing our assumptions... If nothing else, it may allow us to find some previously-unknown scenarios which lead to needing a full format, which would lead to longer test times.

Reply:
as @kjacque said it should no longer damage the media , but I'm not against adding a test into CI